### PR TITLE
Slightly better supervision tree

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: lib/relay/marathon/supervisor.ex [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Supervisor.start_link(__MODULE__, arg, options)
+  end
+
+  def init(_opts) do
+    children = [
+      {Marathon.Store, [name: Marathon.Store]},
+      {Marathon, [name: Marathon]}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
```
Comment (JayH5) [inline lib/relay/marathon/supervisor.ex]: If `Marathon` depends on `Marathon.Store` should this be `:rest_for_one`?

Comment (jerith) [reply lib/relay/marathon/supervisor.ex]: Good catch, thanks. I guess this indicates that we need a new test. :-/

File: test/relay/supervisor_test.exs [hunk 2]
```diff
@@ -278,138 +248,122 @@ defmodule Relay.SupervisorTest do
...
+      # this, we catch exits (which is almost always a terrible idea) and
+      # return `false` instead.
+      :exit, _ ->
+        false
+    end
 
-    # Wait for all the initial interation to finish
-    Process.sleep(50)
+    def wait_until_live do
+      case Enum.any?(@sups, &sup_live?/1) do
```
Comment (JayH5) [inline test/relay/supervisor_test.exs]: Why do we wait for just _any_ supervisor to be live?

Comment (jerith) [reply test/relay/supervisor_test.exs]: 🤦‍♂️  That should be `Enum.all?`. Which probably explains some other weirdness.

**Pull request merged**